chore(lint): add clippy rule to prevent unwrap usage

baerwang opened the pull request:
https://github.com/arana-db/kiwi/issues/218

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Strengthened development build checks to enforce stricter coding standards and promote more robust code practices across the project.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

marsevilspirit commented on cmake/zlib.cmake:
```diff
@@ -13,6 +13,7 @@ ExternalProject_Add(
         GIT_REPOSITORY  "https://github.com/madler/zlib.git"
         GIT_TAG         "v1.2.8"
         GIT_SHALLOW     true
+        UPDATE_COMMAND ""
```
看看这个AI的comment,下面的跟这个重复了

marsevilspirit commented on cmake/zstd.cmake:
```diff
@@ -16,6 +16,7 @@ ExternalProject_Add(
         DOWNLOAD_NAME "zstd-1.5.4.tar.gz"
         SOURCE_DIR ${zstd_SOURCES_DIR}
         DOWNLOAD_NO_PROGRESS 1
+        UPDATE_COMMAND ""
```
看看这个AI的comment,下面的跟这个重复了

marsevilspirit reviewed (commented):
看一下comment

baerwang committed:
chore(lint): add clippy rule to prevent unwrap usage

- Add `-D clippy::unwrap_used` flag to cargo clippy linting command
- Enforce stricter code quality standards by disallowing unwrap calls
- Improve error handling practices across the codebase

lqxhub commented:
@baerwang 麻烦继续处理一下CI流程错误

baerwang commented:
> @baerwang 麻烦继续处理一下CI流程错误

我只负责ci不复杂代码修改

AlexStocks commented on Makefile:
```diff
@@ -53,7 +53,7 @@ fmt-check:
 lint:
 	@echo "Linting code..."
-	@cargo clippy --manifest-path ./Cargo.toml --all-features --workspace -- -D warnings
+	@cargo clippy --manifest-path ./Cargo.toml --all-features --workspace -- -D warnings -D clippy::unwrap_used
```
直接加 `-D clippy::unwrap_used` 而没有先清理代码库里现有的 `unwrap()` 调用，会立刻让 `make lint` 挂掉。这个 lint rule 应该在把现有 unwrap() 全部清理完之后再加，或者先用 `#[allow(clippy::unwrap_used)]` 标注遗留的地方，分批清理。建议在 PR 描述或关联的 issue #218 里说明当前 codebase 里有多少 unwrap() 需要处理，分几个 PR 清理完再合并这个 lint 规则。
